### PR TITLE
Adds support for GPU-enabled function executors to Hydro cluster

### PR DESCRIPTION
**Pull request by vsreekanti**
This doesn't affect Cloudburst yet, which will be added in a separate PR on the Cloudburst repo.

### CODE REVIEW

File: dockerfiles/hydro-base.dockerfile [hunk 1]
```diff
@@ -80,3 +78,8 @@ RUN git clone --recurse-submodules https://github.com/hydro-project/anna
 RUN git clone --recurse-submodules https://github.com/hydro-project/anna-cache
 RUN git clone --recurse-submodules https://github.com/hydro-project/cloudburst
 RUN git clone --recurse-submodules https://github.com/hydro-project/cluster
+
+WORKDIR /hydro/anna
+RUN bash scripts/build.sh -j4 -bRelease
```
Comment (cw75) [inline dockerfiles/hydro-base.dockerfile:83]: why are we now building Anna in the base image?

File: hydro/cluster/create_cluster.py [hunk 2]
```diff
@@ -18,6 +18,7 @@
 import os
 
 import boto3
+import kubernetes
```
Comment (cw75) [inline hydro/cluster/create_cluster.py:21]: Where is this used?

File: hydro/cluster/create_cluster.py [hunk 3]
```diff
@@ -26,15 +27,15 @@
...
+def create_cluster(mem_count, ebs_count, func_count, gpu_count, sched_count,
+                   route_count, bench_count, cfile, ssh_key, cluster_name,
+                   kops_bucket, aws_key_id, aws_key):
 
     if 'HYDRO_HOME' not in os.environ:
         raise ValueError('HYDRO_HOME environment variable must be set to be '
                          + 'the directory where all Hydro project repos are '
                          + 'located.')
-    prefix = os.path.join(os.environ['HYDRO_HOME'], 'cluster/hydro/cluster')
+    prefix = os.path.join(os.environ['HYDRO_HOME'], 'gpu-cluster/hydro/cluster')
```
Comment (cw75) [inline hydro/cluster/create_cluster.py:38]: seems that the prefix that contains "gpu-cluster" is used below to create other nodes as well? Is this what we want?

File: dockerfiles/hydro-base.dockerfile [hunk 4]
```diff
@@ -12,7 +12,7 @@
 #  See the License for the specific language governing permissions and
 #  limitations under the License.
 
-FROM ubuntu:14.04
+FROM ubuntu:18.04
```
Comment (cw75) [inline dockerfiles/hydro-base.dockerfile:15]: wow we finally upgraded from 14.04!

Comment (vsreekanti) [reply dockerfiles/hydro-base.dockerfile:15]: 🙂

File: dockerfiles/hydro-base.dockerfile (hunk 1 above)
Comment (vsreekanti) [reply dockerfiles/hydro-base.dockerfile:83]: Ah, this was a test. I'll remove this.

File: hydro/cluster/create_cluster.py (hunk 3 above)
Comment (vsreekanti) [reply hydro/cluster/create_cluster.py:38]: Nope, this was a mistake. Will remove.

**Pull request merged**